addons/mail/static/src/discuss/core/web/thread_actions.js

```diff
@@ -2,6 +2,22 @@ import { registerThreadAction } from "@mail/core/common/thread_actions";
 
 import { _t } from "@web/core/l10n/translation";
 
+export const joinChannelAction = {
+    condition: ({ owner, thread }) =>
+        thread &&
+        !["chat", "group"].includes(thread.channel_type) &&
+        !thread.self_member_id &&
+        !owner.isDiscussSidebarChannelActions,
```

Comment:
Missing condition to dismiss non-channels: currently the button is also visible in mailboxes (should not be there):

<img width="957" height="277" alt="Image" src="https://github.com/user-attachments/assets/a3cf2a41-96e2-4fad-9bb3-4537936b43c4" />